addon/components/base/bs-carousel.js

```diff
@@ -0,0 +1,580 @@
+   *
+   * @property directionalClassName
+   * @private
+   * @type string
+   */
+  directionalClassName: null,
+
+  /**
+   * Indicates the next slide index to move into.
+   *
+   * @property followingIndex
+   * @private
+   * @type number
+   */
+  followingIndex: null,
+
+  /**
+   * The following slide object that is going to be used by the nested slides components.
+   *
+   * @property followingIndex
+   * @private
+   */
+  followingSlide: computed('childSlides', 'followingIndex', function() {
+    return this.get('childSlides').objectAt(this.get('followingIndex'));
+  }).readOnly(),
+
+  /**
+   * @private
+   * @property hasInterval
+   * @type boolean
+   */
+  hasInterval: gt('interval', 0),
+
+  /**
+   * This observer is the entry point for programmatically slide changing.
+   *
+   * @property indexObserver
+   * @private
+   */
+  indexObserver: observer('index', function() {
+    this.send('toSlide', this.get('index'));
+  }),
+
+  /**
+   * @property indicators
+   * @private
+   */
+  indicators: computed('childSlides.length', function() {
+    return [...Array(this.get('childSlides.length'))];
+  }),
+
+  /**
+   * If user is hovering its cursor on component.
+   * This property is only manipulated when 'pauseOnMouseEnter' is true.
+   *
+   * @property isMouseHovering
+   * @private
+   * @type boolean
+   */
+  isMouseHovering: false,
+
+  /**
+   * The class name to append to the next control link element.
+   *
+   * @property nextControlClassName
+   * @type string
+   * @private
+   */
+  nextControlClassName: null,
+
+  /**
+   * Bootstrap style to indicate the next/previous slide.
+   *
+   * @property orderClassName
+   * @private
+   * @type string
+   */
+  orderClassName: null,
+
+  /**
+   * The current state of the current presentation, can be either "didTransition"
+   * or "willTransit".
+   *
+   * @private
+   * @property presentationState
+   * @type string
+   */
+  presentationState: null,
+
+  /**
+   * The class name to append to the previous control link element.
+   *
+   * @property prevControlClassName
+   * @type string
+   * @private
+   */
+  prevControlClassName: null,
+
+  /**
+   * @private
+   * @property shouldNotDoPresentation
+   * @type boolean
+   */
+  shouldNotDoPresentation: lte('childSlides.length', 1),
+
+  /**
+   * @private
+   * @property shouldRunAutomatically
+   * @type boolean
+   */
+  shouldRunAutomatically: and('autoPlay', 'hasInterval'),
+
+  /**
+   * Starts automatic sliding on page load.
+   * This parameter has no effect if interval is less than or equal to zero.
+   *
+   * @default false
+   * @property autoPlay
+   * @public
+   * @type boolean
+   */
+  autoPlay: false,
+
+  /**
+   * Should have hard stop on corners, i.e., first slide won't make a transition to the
+   * last slide and vice versa.
+   *
+   * @default true
+   * @property continuouslyCycle
+   * @public
+   * @type boolean
+   */
+  continuouslyCycle: true,
```

Comment:
Is there actually a reason this is not named `wrap` as in the original Bootstrap?

Reply:
Many properties aren't named as the originals because when I came across these names for the first time it was **really** hard to figure out what they were actually doing. That's why I've renamed them to something more self-explanatory.

I don't mind at all about changing these names to something else, here are the nomenclature tables defined in the documentation:

 | Description | Original | Component |
 | ------ | ------ | ------ |
 | Autoplays after first user event or on page load. | ride='carousel'\|false | autoPlay=false\|true |
 | Disable automatic cycle. | interval=false | interval=0 |
 | If first slide should follow last slide on "previous" event, the  opposite will also be true for "next" event. | wrap=false\|true | continuouslyCicle=false\|true |
 | Jumps into specific slide index | data-slide-to=n | index=n |
 | Keyboard events. | keyboard=false\|true | keyboard=false\|true |
 | Left-to-right or right-to-left sliding. | N/A |  ltr=false\|true |
 | Pause current cycle on mouse enter. | pause='hover'\|null | pauseOnMouseEnter=false\|true |
 | Show or hide controls  | Tag manipulation. | showControls=false\|true |
 | Show or hide indicators  | Tag manipulation. | showIndicators=false\|true |
 | Waiting time of slides in a automatic cycle. | interval=n | interval=n |

| Operation | Description |
| ------ | ------ |
| Transition | Swaps two slides. |
| Interval | Waiting time after a transition. |
| Presentation | Represents a single transition, or a single interval, or the union of both. |
| Cycle | Presents all slides until it reaches first or last slide. |
| Continuously cycle | Infinite cycling without stops. |